### ISSUE TITLE
Implement spritesheet loading

### ISSUE THREAD

**Issue by nelsonxb** [SourceComb/game-off-2016#4]
The original idea of using sc-sprites, in retrospect, was kind of dumb.

New idea: describe the sprites on the sheet directly in source. Each sheet is described using some kind of DSL (maybe using metaclasses!)

**Comment by nelsonxb** [comment]
Oh god, I've gotten excited about this now.

**Comment by nelsonxb** [comment]
I've gotten an idea for an interface:

``` python
from .sheet import *


class MainSheet(Spritesheet):
    info.canvas = 'assets/player.png'
    info.defaults.scale = 2
    info.defaults.rate = 8


class PlayerSprite(MainSheet):
    idle = Sprite(pos=(0, 0), size=(28, 32))
    run = Sprite(pos=(0, 32), size=(28, 32), frames=8)


class WeaponSprite(MainSheet):
    info.defaults.scale = 1
    axe_static = Sprite(pos=(28, 0), size=(84, 32))
    axe_swing = Sprite(pos=(0, 64), size=(84, 32), frames=4)


################################################################################
from sprite import PlayerSprite


# ...
sprite = PlayerSprite.idle(*spriteArgs)
# ...
```

**Comment by nelsonxb** [comment]
Implemented in 3d610fc8cf34bb69b6cd6b839d4f9ccf8cbd674a. I'm really pleased with how this turned out. See `game/sprite` for implementation and examples.
